### DISCUSSION TITLE
Ejemplos Bots Paramétricos para documentación

### DISCUSSION THREAD

**Discussion by NicolasPLG** [Cotalker/documentation#374] [1. General]
Hola a todos

Dejaré algunos ejemplos de nwr que he usado y puede que @ValeMP le sirva

**Obtener el asset en el momento de crear una tarea**
```
{
                "key" : "get_asset", 
                "name" : "NWRequest", 
                "data" : {
                    "url" : "$JOIN#/#($ENV#BASEURL)#api#properties#($OUTPUT#create_task#task|asset)", 
                    "method" : "GET", 
                    "defaultAuth" : true
                }, 
                "next" : {
                    "SUCCESS" : "", 
                    "ERROR" : ""
                }, 
                "customNetworkRequest" : [

                ]
            },
```
**Obtener la property de un campo del formulario**
```
  {
                "key" : "get_property", 
                "name" : "NWRequest", 
                "data" : {
                    "url" : "$JOIN#/#($ENV#BASEURL)#(api)#(properties)#($VALUE#data|[find=>identifier=cp_client_operaciones]|process|0)", 
                    "method" : "GET", 
                    "defaultAuth" : true
                }, 
                "next" : {
                    "SUCCESS" : "create_project", 
                    "ERROR" : "send_nack"
                }, 
                "customNetworkRequest" : [

                ]
            }
```

**Obtener la información de un usuario**
```
{
                "key" : "get_user", 
                "name" : "NWRequest", 
                "version" : "2.0.0", 
                "data" : {
                    "url" : "$JOIN#/#($ENV#BASEURL)#api#V2#users#($VALUE#user)", 
                    "method" : "GET", 
                    "defaultAuth" : true
                }, 
                "next" : {
                    "SUCCESS" : "get_center", 
                    "ERROR" : ""
                }, 
                "customNetworkRequest" : [

                ]
            }
```

Editaré a medida que los encuentre. Estaba pensando en colocar combinaciones de estos, pero sería en otra discussion asociando a ésta.

Gracias =>

**Comment by acedward** [comment]
Hola @NicolasPLG

Sería bueno agregar en cada ejemplo el $CODE equivalente,
Sólo para que no se use NWRequest custom cuando se puede utilizar un $CODE prefabricado

```
Ejemplo 1: 
$CODE#properties#id#($OUTPUT#create_task#task|asset)

Ejemplo 2:
$CODE#properties#id#($VALUE#data|[find=>identifier=cp_client_operaciones]|process|0)

Ejemplo 3
$CODE#user#id#($VALUE#user)
```


Referencia:


![image](https://user-images.githubusercontent.com/3951895/130699664-56c10d1f-1cdc-4f7e-a412-4d18f4dc7582.png)

**Reply by NicolasPLG** [reply]
Hola @acedward muchas gracias por tu apoyo y respuesta. trataré de agregar el code equivalente, pero con lo que colocaste está muy bueno.

**Reply by acedward** [reply]
Por otra parte hace poco sentido 

`($VALUE#data|[find=>identifier=cp_client_operaciones]|process|0)`
`$OUTPUT#create_task#task|asset`

 sin el contexto correcto.
¿Cómo podemos mejorar esto?

**Comment by acedward** [comment]
¿Dejamos esta discusión abierta para agregar más ejemplos?

**Reply by NicolasPLG** [reply]
tienes razón así se llena la biblioteca.

Buscaré si tengo también + de 2 rutinas para colocar los ejemplos

**Comment by ValeMP** [comment]
Hola!

Hice una tabla de ideas de rutinas y cotlang que se podrían agregar a la documentación. Se podría seguir la misma historia de los tutoriales y agregarlos ahi. Se podría aprovechar que además de explicar cómo configurar la rutina, se podría explicar cómo se llego al cotlang que se usa.

<google-sheets-html-origin style="font-style: normal; font-variant-caps: normal; font-weight: normal; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none; caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0);">

Rutinas
----
|PbBot | Contexto | Descripción | COTLANG|
|-------|----------|-------------|-----------|
|Network Request | bot | Realizar un Network Request externoy luego usar OUTPUT para obtener la respuesta | OUTPUT|
|Network Request | bot | Realizar un Network Request a cotalker y luego usar OUTPUT para obtener la respuestas y realizar alguna acción | OUTPUT, JOIN, ENV|
| Invitar a un usuario al canal | survey change state | Realizar una invitación por access role |  |
|Crear Tarea | start workflow | Crear una tarea que sea obteniendo respuetas del forms que ademas esten con aditional file y asignee opcional | VALUE, CODE|
|enviar email | survey trigger | explicando que puede ir en cada field y que sea a traves de recurrir info de un cliente que es una propiedad | CODE, VALUE|
|switch | survey trigger | dependiendo de la respuesta, deriva a enviar un email o integracion con wpp | VALUE, CODE|
|Actualizar tarea | start workflow | Actualizar el serial de la tarea | OUTPUT|
|Integración con Whatsapp | survey trigger | (No lo he usado, pero podría ser interesante de agregar)| |

</google-sheets-html-origin>

Cuéntenme su opinión para poder buscar y entregar ejemplos en Cotalker.

Además, tengo este material que hice para los partner que se puede agregar a la documentación. Es un "torpedo". Avísenme con qué otras cosas podría agregar o cambiar. 
[Cheat sheet-3.pdf](https://github.com/Cotalker/documentation/files/7060600/Cheat.sheet-3.pdf)

Al final la info que esta en este torpedo esta en la documentación, pero dejé algunas cosas que encontré que podrían servir como resumen. También vi que lo hacía Vue y me sirvio harto cuando estaba aprendiendo.

**Reply by acedward** [reply]
¿Quizás cada rutina debería tener uno o más ejemplos?
https://doc.cotalker.com/docs/documentation/automation/existing_routines

**Reply by ValeMP** [reply]
me parece una buena oportunidad! Solo me preocupa que si se quiere hacer un ejemplo de cada uno rápido, se pierda la explicación del cotlang. Se podría dar prioridad a las rutinas más usadas con explicación larga y el resto podría ser con explicación no tan profunda.
